src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const incomingData = [...this];
+
+    if (incomingData.length > 0) {
+      let result = '';
+
+      for (let i = 0; i < incomingData.length; i++) {
```

Comment:
Don't write loops inside if. It's making your code too complicated.
Remember that loop with false condition simply will iterate 0 times.